### PR TITLE
feat: add rocketmq binding

### LINKED ISSUE #494: Integration with Apache RocketMQ
Hi, I am a contributor of Apache RocketMQ, since RocketMQ has been merge into the [proprietary part of the spec](https://github.com/cloudevents/spec/blob/main/cloudevents/proprietary-specs.md), could we provide a RocketMQ protocol binding implementation?

### PR DESCRIPTION
**Pull request by aaron-ai**
Fixes #494

### CODE REVIEW

**Comment by aaron-ai** [comment]
@pierDipi Could you please take a look?

File: rocketmq/src/main/java/io/cloudevents/rocketmq/RocketMqMessageFactory.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    /**
+     * Creates a {@link MessageReader} to read a RocketMQ {@link MessageView}.
+     *
+     * @param message The RocketMQ {@link MessageView} to read from.
+     * @return A {@link MessageReader} that can read the given {@link MessageView} to a {@link io.cloudevents.CloudEvent} representation.
+     */
+    public static MessageReader createReader(final MessageView message) {
+        final ByteBuffer byteBuffer = message.getBody();
+        byte[] body = new byte[byteBuffer.remaining()];
+        byteBuffer.get(body);
```
[the remark this answers is not in the document]
Comment (aaron-ai) [reply rocketmq/src/main/java/io/cloudevents/rocketmq/RocketMqMessageFactory.java:51]: Usually the `ByteBuffer` returned is readonly, so we have to copy it.

**Comment by aaron-ai** [comment]
@pierDipi Thanks for you review! Could you help to re-run the workflow?

**Review by pierDipi — COMMENTED** [review]
I have another suggestion, can we use a single package instead of the `impl` package? I know some components in the SDK are using that approach (I'd like to move away from that), however, that forces us to make some APIs public where they could be package-private.

File: rocketmq/src/main/java/io/cloudevents/rocketmq/impl/RocketmqConstants.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    /**
+     * The prefix name for CloudEvent attributes for use in properties of a RocketMQ message.
+     */
+    public static final String CE_PREFIX = "CE_";
+
+    public static final Map<String, String> ATTRIBUTES_TO_PROPERTY_NAMES = MessageUtils.generateAttributesToHeadersMapping(CEA -> CE_PREFIX + CEA);
+
+    public static final String PROPERTY_CONTENT_TYPE = "CE_contenttype";
+    public static final String MESSAGE_PROPERTY_SPEC_VERSION = ATTRIBUTES_TO_PROPERTY_NAMES.get("specversion");
+}
```
[the remark this answers is not in the document]
Comment (aaron-ai) [reply rocketmq/src/main/java/io/cloudevents/rocketmq/impl/RocketmqConstants.java]: Fixed already.

**Review by pierDipi — APPROVED** [review]
Thanks @aaron-ai!

LGTM

**Pull request merged**